website/static/js/components/autocomplete.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * Components involving the typeahead.js autocomplete library.
+ *
+ * Inspired by the onboarder component, but refactored into a generic view-model
+ * that can be subclassed to create components to search through lists of
+ * objects other than nodes.
+**/
+'use strict';
+var $ = require('jquery');
+var ko = require('knockout');
+var $osf = require('js/osfHelpers');
+require('typeahead.js');
+
+var MAX_RESULTS = 14;
```

Comment:
As discussed in person, this seems like a very arbitrary limit, and can be revisited.